src/hypertable.c

```diff
@@ -964,10 +965,17 @@ hypertable_insert(int32 hypertable_id, Name schema_name, Name table_name,
 	{
 		NameData default_associated_table_prefix;
 		memset(NameStr(default_associated_table_prefix), '\0', NAMEDATALEN);
-		snprintf(NameStr(default_associated_table_prefix),
-				 NAMEDATALEN,
-				 DEFAULT_ASSOCIATED_TABLE_PREFIX_FORMAT,
-				 fd.id);
+		Assert(replication_factor >= 0);
+		if (replication_factor == 0)
+			snprintf(NameStr(default_associated_table_prefix),
```

Comment:
I suggest we add a check for return value/length here and raise an error in case the name was truncated, like we do for the chunk table name above.

Reply:
Also nit, maybe call `snprintf` just once in that case but place naming change logic under if here? Likely will require an additional variable